File: codechef/contests/NOV14/PRPALIN.cpp
#include <bits/stdc++.h>

using namespace std;

bool ispal(string a){
	string b(a.rbegin(), a.rend());

//	cerr << "a: " << a << " rev: " << b << endl;
	return a == b;
}

bool solve(string a){
	int n = a.length();
	if (ispal(a)) return true;

	for(int i = 0; i < n/2; i += 1){
		int j = n-i-1;
		if (a[i] != a[j]){
			string b(a.begin(), a.end());
			a.erase(a.begin()+i, a.begin()+i+1);
			b.erase(b.begin()+j, b.begin()+j+1);
//			cerr << a << " " << b << endl;	
		
			return ispal(a) or ispal(b);
		}
	}

	return false;
}


int main(void){
	int t;
	cin >> t;
	while(t--){
		string a;
		cin >> a;
		cout << (solve(a) ? "YES" : "NO") << endl;
	}
}

File: codechef/problems/PRPALIN.py
def ispal(a):
	return a == a[::-1]

def solve(a):

	a = list(a)
	if ispal(a): return 1
	for i in xrange(len(a)/2):
		j = len(a) -i- 1 
		if a[i] != a[j]:
			b = a[:]
			del a[i]; del b[j]
			return ispal(a) or ispal(b)
	return 1

for _ in xrange(input()):
	print "YES" if solve(raw_input()) else  "NO"
